File: server/src/main/java/furman/pay/conroller/DayOrderService.java
package furman.pay.conroller;

import furman.pay.model.PayOrder;
import furman.pay.model.day.Day;
import furman.pay.model.day.DayOrder;
import furman.pay.model.day.QDayOrder;
import furman.pay.repository.PayOrderRepository;
import furman.pay.repository.WorkRepository;
import furman.pay.repository.day.DayOrderRepository;
import furman.pay.repository.day.DayRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.querydsl.QSort;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.Iterator;

/**
 * akoiro - 10/30/15.
 */
@RestController
public class DayOrderService {
    @Autowired
    private DayOrderRepository dayOrderRepository;

    @Autowired
    private WorkRepository workRepository;

    @Autowired
    private PayOrderRepository payOrderRepository;

    @Autowired
    private DayRepository dayRepository;

    @RequestMapping("/dayOrderService/deleteByDay")
    public void deleteByDay(Day day) {
        if (day.getId() != null) {
            Iterable<DayOrder> dayOrders = dayOrderRepository.findAll(QDayOrder.dayOrder.day.eq(day));
            dayOrderRepository.delete(dayOrders);
        }
    }

    @RequestMapping("/dayOrderService/countForDay")
    public long countForDay(@RequestParam(required = true) String dayId) {
        Day day = dayRepository.findOne(dayId);
        return dayOrderRepository.count(QDayOrder.dayOrder.day.eq(day));
    }

    @RequestMapping("/dayOrderService/findAllForDay")
    public Iterable<DayOrder> findAllForDay(@RequestParam(required = true) String dayId) {
        Day day = dayRepository.findOne(dayId);
        Iterable<DayOrder> dayOrders = dayOrderRepository.findAll(QDayOrder.dayOrder.day.eq(day), QDayOrder.dayOrder.payOrder.number.asc());
        return dayOrders;
    }


    @RequestMapping("/dayOrderService/createDayOrder")
    public DayOrder createDayOrder(@RequestParam(required = true) String dayId, @RequestParam String payOrderId) {
        DayOrder dayOrder = new DayOrder();
        dayOrder.setDay(dayRepository.findOne(dayId));
        dayOrder.setPayOrder(payOrderRepository.findOne(payOrderId));

        dayOrder.setOrderValues(
                CollectOrderValues.valueOf(dayOrder.getPayOrder(),
                        getFirstDayOrderFor(dayOrder.getPayOrder()),
                        workRepository.findAll()).collect());
        dayOrderRepository.insert(dayOrder);
        return dayOrder;
    }

    private DayOrder getFirstDayOrderFor(PayOrder payOrder) {
        DayOrder firstDayOrder = null;
        Iterator<DayOrder> iterator = dayOrderRepository.findAll(QDayOrder.dayOrder.payOrder.eq(payOrder), new QSort(QDayOrder.dayOrder.created.asc())).iterator();
        if (iterator.hasNext()) {
            firstDayOrder = iterator.next();
        }
        return firstDayOrder;
    }
}

File: client/src/main/resources/public/app/service/DayOrderService.js
'use strict';

angular.module('app.service').factory('dayOrderService', DayOrderService);

function DayOrderService($http, dayOrderRepository) {
    var baseUrl = "/api/pay/dayOrderService";

    var service = {};

    service.deleteByDay = deleteByDay;
    service.createDayOrder = createDayOrder;
    service.findAllForDay = findAllForDay;
    service.countForDay = countForDay;
    service.update = update;

    function deleteByDay(day) {
        $http.put(baseUrl + "/deleteByDay", day);
    }

    function createDayOrder(day, payOrder) {
        return $http.get(baseUrl + "/createDayOrder?dayId=" + day.id + "&payOrderId=" + payOrder.id);
    }

    function findAllForDay(day) {
        return $http.get(baseUrl + "/findAllForDay?dayId=" + day.id);
    }

    function countForDay(day) {
        return $http.get(baseUrl + "/countForDay?dayId=" + day.id);
    }

    function update(dayOrder) {
        return dayOrderRepository.update(prepareToSave(dayOrder));
    }

    function prepareToSave(dayOrder) {
        var toSave = jQuery.extend(true, {}, dayOrder);

        toSave.day = "/api/pay/day/" + dayOrder.day.id;
        toSave.payOrder = "/api/pay/payOrder/" + dayOrder.payOrder.id;

        toSave.orderValues.forEach(function (orderValue) {
            orderValue.work = "/api/pay/work/" + orderValue.work.id;
        });
        return toSave;
    }

    return service;
}
